lib/python/ray/worker.py

```diff
@@ -554,34 +574,37 @@ def func_executor(arguments):
     def func_call(*args, **kwargs):
       """This gets run immediately when a worker calls a remote function."""
       args = list(args)
-      args.extend([kwargs[keyword] if kwargs.has_key(keyword) else default for keyword, default in func_call.keyword_defaults[len(args):]]) # fill in the remaining arguments
-      if worker.mode == ray.PYTHON_MODE:
+      args.extend([kwargs[keyword] if kwargs.has_key(keyword) else default for keyword, default in keyword_defaults[len(args):]]) # fill in the remaining arguments
+      if _mode() == ray.PYTHON_MODE:
         # In ray.PYTHON_MODE, remote calls simply execute the function. We copy
         # the arguments to prevent the function call from mutating them and to
         # match the usual behavior of immutable remote objects.
         return func(*copy.deepcopy(args))
-      check_arguments(func_call, args) # throws an exception if args are invalid
-      objrefs = worker.submit_task(func_call.func_name, args)
+      check_arguments(arg_types, has_vararg_param, func_name, args) # throws an exception if args are invalid
+      objrefs = _submit_task(func_name, args)
```

Comment:
Actually, then it will try to pickle `worker`, and `worker` can't be pickled.
```
TypeError: can't pickle PyCapsule objects
```

I agree that that is a problem though. Let me think about it..